routing/routing.go

```diff
@@ -175,17 +225,59 @@ func (r *Routing) startReceivingUpdates(o Options) {
 	}()
 }
 
-// Matches a request in the current routing tree.
+// Route matches a request in the current routing tree.
 //
 // If the request matches a route, returns the route and a map of
 // parameters constructed from the wildcard parameters in the path
 // condition if any. If there is no match, it returns nil.
 func (r *Routing) Route(req *http.Request) (*Route, map[string]string) {
-	m := r.matcher.Load().(*matcher)
-	return m.match(req)
+	rt := r.routeTable.Load().(*routeTable)
+	return rt.m.match(req)
 }
 
-// Closes routing, stops receiving routes.
+// Close closes routing, stops receiving routes.
 func (r *Routing) Close() {
 	close(r.quit)
 }
+
+func slice(r []*eskip.Route, offset int, limit int) []*eskip.Route {
+	if offset > len(r) {
+		offset = len(r)
+	}
+	end := offset + limit
+	if end > len(r) {
+		end = len(r)
+	}
+	result := r[offset:end]
+	if result == nil {
```

Comment:
figured.. when i am trying just simply in go it seems like go-pry doesn't work exactly the same way:

```
var a []int
println(a == nil, a[0:0] == nil)
// Output: true true
```

and:

```
var a = []string{"foo"}
println(a[0:0] == nil, a[1:1] == nil)
// Output: false false
```

The reason to do this nil check, I assume, is to avoid sending null as the json response instead of an empty array in cases when the input routes are nil for any reason.